### PR TITLE
Fixes #1453

### LINKED ISSUE #1453: If call @model.file.url() on a non saved istance get a url instead of nil
Hi,

In my action controller call `render 'action'` if model validations fails.

In the `action` view have
```
<div id="preview" data-preview-url="<%= @model.preview.url %>">
```
that return good url if `@model` is saved on db, bad url without :`id` part(obviously) if `@model` is not stored to db. Have to return `nil` in this case or I'm wrong?

### PR DESCRIPTION
**Pull request by byterussian**

### CODE REVIEW

**Comment by maclover7** [comment]
Hi @byterussian ! Can you edit your tests to make sure everything is passing? Thanks!

File: spec/paperclip/storage/s3_spec.rb [hunk 1]
```diff
@@ -419,11 +429,13 @@ def original_filename
         production: { bucket: "prod_bucket" },
         development: { bucket: "dev_bucket" }
       },
+      bucket: "bucket",
```
Comment (byterussian) [inline spec/paperclip/storage/s3_spec.rb]: Anyone can help me with this? Maybe:
```
      rebuild_model storage: :s3,
                    s3_credentials: {
                      production: { bucket: "prod_bucket" },
                      development: { bucket: "dev_bucket" }},
                    bucket: "bucket",
                    s3_host_alias: "something.something.com",
                    path: ":attachment/:basename:dotextension",
                    url: ":s3_alias_url"
      @dummy = Dummy.new
      @dummy.avatar = stringy_file
      @dummy.stubs(:new_record?).returns(false)
```

**Comment by byterussian** [comment]
Hi @maclover7, now all the tests pass.

**Comment by jyurek** [comment]
@byterussian If you could squash these commits together, I'll gladly pull this in. Thanks!

**Pull request closed without merging**